### PR TITLE
Fix after demo

### PR DESCRIPTION
**Pull request by guragt**
1. For users without provider will be shown 'native' in views.
2. API: Users_controller#show return 404 if user does not exist
3. Mailer for reset okta password instructions

Screenshot 2021-05-28 at 12 13 34
Screenshot 2021-05-28 at 12 15 11

### CODE REVIEW

File: app/views/admin/users/index.html.erb [hunk 1]
```diff
@@ -29,7 +29,7 @@
             <td><%= user.likes.count %></td>
             <td><%= user.private? ? t('.private') : t('.public') %></td>
             <td><%= user.deleted_at? ? t('.deleted') : t('.active') %></td>
-            <td><%= user.provider %></td>
+            <td><%= user.provider? ? user.provider : t('.native') %></td>
```
Comment (Snick555) [inline app/views/admin/users/index.html.erb]: `user.provider || t('.native')`

File: app/views/api/v1/users/_user.json.jbuilder [hunk 2]
```diff
@@ -2,4 +2,8 @@ json.id user.id
 json.name user.name
 json.nickname user.nickname
 json.email user.email
-json.provider user.provider
+if user.provider?
```
Comment (Snick555) [inline app/views/api/v1/users/_user.json.jbuilder]: `json.provider user.provider || t('.native')`

File: app/controllers/api/v1/users_controller.rb [hunk 3]
```diff
@@ -8,15 +8,16 @@ def index
...
+        render json: 'Not Found', status: :not_found unless @user
       end
 
       private
 
       def check_provider
         return if params[:provider] == 'native' || params[:provider] == 'oktaoauth'
 
-        render json: t('.bad_request'), status: :bad_request
+        render json: 'Bad request', status: :bad_request
```
Comment (Snick555) [inline app/controllers/api/v1/users_controller.rb]: I'd prefer to leave locale

File: app/views/admin/users/index.html.erb [hunk 4]
```diff
@@ -29,7 +29,7 @@
             <td><%= user.likes.count %></td>
             <td><%= user.private? ? t('.private') : t('.public') %></td>
             <td><%= user.deleted_at? ? t('.deleted') : t('.active') %></td>
-            <td><%= user.provider? ? user.provider : t('.native') %></td>
+            <td><%= user.provider || t('.native') %></td>
```
Comment (boikonataliii) [inline app/views/admin/users/index.html.erb]: you can move it to user method

File: config/locales/en.yml [hunk 5]
```diff
@@ -111,6 +120,12 @@ en:
       unlocked: "Unlocked"
     gallery:
       no_posts: "No posts yet"
+    password_mailer:
+      reset_okta_password_instructions:
+        subject: "Reset password instructions"
+        greeting: "Hello %{email}!"
+        content: "Someone has requested changing your password. You can do this in your"
```
Comment (boikonataliii) [inline config/locales/en.yml]:
```
content_html: "Someone has requested changing your password. You can do this in your <a href="%{okta_url}" target="_blank">okta account."
content_text: "Someone has requested changing your password. You can do this in your okta account %{okta_url} ."
```

**Review by Snick555 — APPROVED** [review]

File: app/views/admin/users/index.html.erb (hunk 4 above)
Comment (guragt) [reply app/views/admin/users/index.html.erb]: But I can't use method "translate" in model

Comment (boikonataliii) [reply app/views/admin/users/index.html.erb]:
```
def human_provider
  provider || I18n.t('.native')
end
```

**Review by boikonataliii — APPROVED** [review]

**Pull request merged**